docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,511 @@
+
+To create your own modular resource, code a module in Go or Python using the module support libraries provided by [Viam's SDKs](/program/apis/) that implements at least one new model or type of {{< glossary_tooltip term_id="resource" text="resource" >}}:
+
+{{< tabs >}}
+{{% tab name="Define a New Model of a Built-In Resource Type" %}}
+
+1. [Code a new resource model](#code-a-new-resource-model) implementing all methods the Viam RDK requires in the API definition of its built-in type (ex. `rdk:component:base`).
+Import your custom model and API into the main program and register the new resource model with your chosen SDK's helper methods.
+
+2. [Code a main program](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
+
+{{% /tab %}}
+{{% tab name="Define a New Type of Resource" %}}
+
+1. Define the messages and methods of the new API in [protobuf](https://github.com/protocolbuffers/protobuf), then [generate code](https://grpc.io/docs/languages/python/generated-code/) in Python or Go and use the generated code to implement the higher level server and client functions required.
+
+2. [Code at least one model](#code-a-new-resource-model) of this new resource.
+Make sure to implement every method required in your API definition.
+Import your custom models and APIs into the main program and register the new resource models with your chosen SDK's helper methods.
+
+3. [Code a main program](#code-a-main-program-that-serves-as-the-entry-point-file-to-the-module) that starts the module after adding your desired resources from the registry.
+This main program is the "entry point" to your module.
+
+4. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
+
+{{% /tab %}}
+{{% /tabs %}}
+
+### Code a new resource model
+
+The following example module registers a modular resource implementing Viam's built-in [Base API](/components/base/#api) [(rdk:service:base)](/extend/modular-resources/key-concepts/#models) as a new model, `"mybase"`:
+
+The Go code for the custom model (<file>mybase.go</file>) and module entry point file (<file>main.go</file>) is adapted from the full demo modules available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources).
+
+[Go Client SDK](https://pkg.go.dev/go.viam.com/rdk):
+
+<file>mybase.go</file>
+
+{{%expand "Click to view example code from a Go module implementing a custom model of the base component built-in resource, mybase, and registering the new model and API" %}}
```

Comment:
```suggestion
{{%expand "Click to view example code from a Go module implementing a custom model of the base component built-in resource, mybase, and registering the new model" %}}
```
I think this could confuse people and make people think they're creating a new API instead of implementing an existing one

Reply:
Maybe should add the helpers clarification?

Reply:
like 
```suggestion
{{%expand "Click to view example code from a Go module implementing a custom model of the base component built-in resource, mybase, and registering the new model and API helpers with the client SDK" %}}
```
Struggling as it's a little long but lmk what you think. can incorporate this feedback either way when I get time to rework the whole section re feedback too

Reply:
I'm not sure what the "helpers" adds? I think shorten it as much as possible, and if you have to you can include other details inside the expander or in the body text. See also my formatting suggestion though. And, if you come up with an entirely different format, this could become moot anyway? But if not, maybe:
```suggestion
{{%expand "Click to view example code from a Go module implementing and registering a custom model of the base component" %}}
```

Reply:
done, and added details!